cordyceps/cordyceps/vs_controller.py

```diff
@@ -0,0 +1,120 @@
+import rclpy
+from rclpy.node import Node
+import numpy as np
+import matplotlib.pyplot as plt
+from geometry_msgs.msg import Twist
+import threading
+from .Robot import Robot
+from std_srvs.srv import Trigger
+from cordyceps_interfaces.msg import RobotPaths
+from cordyceps_interfaces.srv import Controller, CheckThread
+
+class ControllerService(Node):
+    def __init__(self, fleet_size=4):
+        super().__init__("cordyceps_controller")
+        self.start_follow_path_service = self.create_service(Controller, "start_follow_path", self.start_thread_callback)
+        self.check_thread_state = self.create_service(CheckThread, "check_thread_state", self.check_thread_state_callback)
+
+        # Constants delcaration
+        self.MAX_BOT_SPEED =  0.5 #m/s
+        self.GOAL_RADIUS = 0.05 #m
+
+        self.robots = []
+        for i in range(fleet_size):
+            self.robots.append(Robot(0,0,0,f"r{i}", self))
+
+        self.follow_paths_thread = None
+
+
+    def start_thread_callback(self, request, response):
+        """Recieves each bot's path and sends the velocities to follow them"""
+
+        self.get_logger().info("Executing goal...")
+        
+        paths = []
+        for i, path in enumerate(request.robot_paths.paths):
+            paths.append([])
+            for poses in path.robot_poses[:]:
+                paths[i].append([poses.x, poses.y])
+
+        self.plot_path(paths)
+        self.follow_paths_thread = threading.Thread(target=self.follow_paths, args=(paths,))
+        self.follow_paths_thread.start()
+        return response
+    
+    def check_thread_state_callback(self, request, response):
+        """Checks if the thread is still running"""
+        response.is_alive = self.follow_paths_thread.is_alive()
+        return response
+
+    def plot_path(self, robot_paths:list[list[tuple[float, float]]]) -> None:
+        labels = []
```

Comment:
```suggestion
    def plot_path(self, robot_paths:list[list[tuple[float, float]]]) -> None:
    """Displays the current path"""
    
```

Reply:
labels[] needs to stay there because it is used in legend